lib/src/tag/tagger.dart

```diff
@@ -355,6 +350,53 @@ class Tagger {
     }
   }
 
+  void swiftPackageManagerPluginTag(
+      List<String> tags, List<Explanation> explanations) {
+    if (!_usesFlutter) return;
+    final pubspec = _pubspecCache.pubspecOfPackage(packageName);
+
+    bool pathExists(dynamic m, List<String> path) {
+      dynamic current = m;
+      for (final e in path) {
+        if (current is! Map) return false;
+        if (!current.containsKey(e)) return false;
+        current = current[e];
+      }
+      return true;
+    }
+
+    var isDarwinPlugin = false;
+    var swiftPmSupport = true;
+
+    for (final darwinOs in ['macos', 'ios']) {
+      if (pathExists(
+          pubspec.originalYaml, ['flutter', 'plugin', 'platforms', darwinOs])) {
+        isDarwinPlugin = true;
+        final specificPackageSwiftFile =
+            path.join(darwinOs, packageName, 'Package.swift');
+        final genericPackageSwiftFile =
+            path.join('darwin', packageName, 'Package.swift');
```

Comment:
> Does `ffiPlugin` only affect method channels, and not affect the need for a podspec or `Package.swift`? (I'm not sure entirely sure, just guessing)

From my understanding, a plugin can support CocoaPods & SwiftPM and also be an `ffiPlugin`. For example: https://docs.flutter.dev/platform-integration/ios/c-interop#open-source-third-party-library

cc @dcharkes to confirm.

> Is `sharedDarwinSource: true` the only circumstance under which we should look in `darwin/<packageName>/Package.swift` ? (I'm suspecting so)

Yup that's correct

> It would probably be useful with Flutter plugin reference documentation.
> Similar to [pubspec reference docs](https://dart.dev/tools/pub/pubspec) covering the properties under `flutter.plugin` and what files they imply exist.

This is an excellent idea. I filed this issue: https://github.com/flutter/website/issues/11334

Reply:
@sigurdm should we take `sharedDarwinSource` into account?